.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -2,12 +2,15 @@
 
 ## PR Checklist
 
-- [ ] Has Pytest style unit tests
-- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
-- [ ] New features are documented, with examples if plot related
-- [ ] Documentation is sphinx and numpydoc compliant
-- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
-- [ ] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way
+<!-- Please mark any checkboxes that do not apply to this PR as [N/A]. -->
+
+- [ ] Has pytest style unit tests (and `pytest` passes).
+- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
+- [ ] New features are documented, with examples if plot related.
+- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
+- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
+- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
+- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
```

Comment:
Does it make sense to organize these a bit?  Also, I've never run `flake8 --docstring-convention=all` in my life.  Is this really a requirement?  

```suggestion
<!-- Please mark any checkboxes that do not apply to this PR as [N/A]. -->

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
    - [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
   - [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
  - [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
```

Reply:
> Does it make sense to organize these a bit? 

I accepted your reorg because it doesn't change anything that anyone would object to, but any further reorg should go into a new PR (feature creep is why my last one died).

> Also, I've never run `flake8 --docstring-convention=all` in my life. Is this really a requirement?

For the first two weeks or so of me submitting PRs, the majority of comments on my PRs were from @timhoffm, correcting issues that would have been caught by running this command.

If you're already versed in our style guide, then this may not feel necessary, but this command will catch all of that stuff automatically, and I think it will save significant review time to have it here.

Reply:
Suggest just link the style conventions rather than suggest a specific way of checking adherence. We don't run these tests ourselves, so its pretty confusing to include here.

Reply:
A lot of things are ignored right now, but flake8 linting does run these tests in this manner.

Reply:
... learn something new every day!